### PR TITLE
[processing] restore most of terrain analysis tools

### PR DESCRIPTION
**Pull request by alexbruy**
## Description
Restore almost all terrain analysis tools (only Relief is missed)

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: python/plugins/processing/algs/qgis/Hillshade.py [hunk 1]
```diff
@@ -58,16 +58,18 @@ def __init__(self):
...
-                                          self.tr('Vertical angle'), 1.00, 90.00, 40.00))
-        self.addOutput(OutputRaster(self.OUTPUT_LAYER,
-                                    self.tr('Hillshade')))
+        self.addParameter(QgsProcessingParameterRasterLayer(self.INPUT,
+                                                            self.tr('Elevation layer')))
+        self.addParameter(QgsProcessingParameterNumber(self.Z_FACTOR,
+                                                       self.tr('Z factor'), QgsProcessingParameterNumber.Double,
+                                                       1, False, 1, 999999.99))
+        self.addParameter(QgsProcessingParameterNumber(self.AZIMUTH,
+                                                       self.tr('zimuth (horizontal angle)'), QgsProcessingParameterNumber.Double,
```
Comment (nyalldawson) [inline python/plugins/processing/algs/qgis/Hillshade.py]: Minor typo here - the "a" from azimuth has been dropped

**Review by nyalldawson — APPROVED** [review]
Looks perfect - just the little typo is the only thing missing!

**Comment by nyalldawson** [comment]
Awesome :)

**Comment by alexbruy** [comment]
Now fighting with Relief and network analysis :-)

**Pull request merged**